.github/workflows/pr.yml

```diff
@@ -14,49 +14,160 @@ on:
   pull_request:
     branches: [ main ]
 
+env:
+  GPG_MAIL: 'infrastructure-eng@newrelic.com'
+  OHAI_PFX_CERTIFICATE_BASE64: ${{ secrets.OHAI_PFX_CERTIFICATE_BASE64 }} # base64 encoded
+  OHAI_PFX_PASSPHRASE:  ${{ secrets.OHAI_PFX_PASSPHRASE }}
+  GORELEASER_VERSION: 'v1.13.1'
+
 ###
 ###   Checking PR # Checks the file and the info provided when an exporter is modified
 ###
 name: Checking PR
 jobs:
-  checking_pr:
-    name: Checking PR
+  check_exporter_preconditions:
+    name: Test exporter e2e and output exporter variables
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-      - uses: actions/setup-go@v2
+      - uses: actions/setup-go@v3
         with:
           go-version-file: nri-config-generator/go.mod
-      - name: Check if exporters have been modified and if so run checks
+      - name: Check if exporters have been modified
         id: check
+        # Running shouldDoRelease we check if any exporter.yml has been modified, if so we set CREATE_RELEASE to true setting the variable EXPORTER_PATH
+        # That variable is used by the following step and jobs
         run : |
           source ./scripts/common_functions.sh
           shouldDoRelease
           loadVariables $EXPORTER_PATH
-          checkExporter
           setStepOutput
-          if [ "$ERRORS" != "" ];then
-            echo Following errors have been found: $ERRORS
-            exit 1
-          fi
+    outputs:
+      CREATE_RELEASE: ${{ steps.check.outputs.CREATE_RELEASE }}
+      EXPORTER_PATH: ${{ steps.check.outputs.EXPORTER_PATH }}
+      PACKAGE_LINUX: ${{ steps.check.outputs.PACKAGE_LINUX }}
+      PACKAGE_WINDOWS: ${{ steps.check.outputs.PACKAGE_WINDOWS }}
+      INTEGRATION_NAME: ${{ steps.check.outputs.NAME }}
+      INTEGRATION_VERSION: ${{ steps.check.outputs.VERSION }}
+      RELEASE_TAG: ${{ steps.check.outputs.PACKAGE_NAME }}-${{ steps.check.outputs.VERSION }}
+      EXPORTER_REPO_URL: ${{ steps.check.outputs.EXPORTER_REPO_URL }}
+      EXPORTER_HEAD: ${{ steps.check.outputs.EXPORTER_HEAD }}
+      EXPORTER_CHANGELOG: ${{ steps.check.outputs.EXPORTER_CHANGELOG }}
```

Comment:
I see some variables output here that are not used by the other steps, like EXPORTER_PATH, but they are in the Prerelease action where this same step is done, I assume is to have a common step, isn't it?

Reply:
There are three reasons why this is this way:
 * First of all, scripts are obscure and from the workflow you are not able to see all [script outputs](https://github.com/newrelic/newrelic-prometheus-exporters-packages/blob/a6c8a2694cb3108f405f38a4f1631daff2633305/scripts/common_functions.sh#L33-L50) so I preferred to leave it as is.
 * That step is a copy and paste from the prerelease because I wanted to replicate the prerelease completely except for the part of uploading the artifacts.
 * I wanted to make the least amount of changes so the refactor does not go out of our hands. The repository needs a refactor from the ground up but this is neither the PR nor the time to do it.

There are my reason but If you see that this is adding more noise to the pipeline I'd be happy to remove anything unused.